src/components/ArticleForm.jsx

```diff
@@ -2,24 +2,36 @@ import React, { useState } from "react";
 import { Form, Container, Checkbox, Segment, Message } from "semantic-ui-react";
 import Article from "../modules/articles";
 import { useHistory } from "react-router-dom";
+import toBase64 from "../modules/toBase64";
 
 const ArticleForm = () => {
   const [message, setMessage] = useState("");
   const [selectedCategory, setSelectedCategory] = useState("");
+  const [image, setImage] = useState();
   const history = useHistory();
 
+  const selectImage = (e) => {
+    setImage(e.target.files[0]);
+  };
+
   const handleCategoryChange = (value) => {
     setSelectedCategory(value);
   };
 
   const onSubmit = async (e) => {
     e.preventDefault();
+    let { title, teaser, content, premium, encodedImage } = e.target;
+
+    if (image) {
+      encodedImage = await toBase64(image);
+    }
     const result = await Article.create(
-      e.target.title.value,
-      e.target.teaser.value,
-      e.target.content.value,
+      title,
+      teaser,
+      content,
       selectedCategory,
-      e.target.premium.checked
+      premium,
```

Comment:
Did you check that this take true/false values with the new way of writing =e.target?